### PR TITLE
Fix(core) incorrect quantity adjustment

### PR DESCRIPTION
**Pull request by WilliamMilne**
Fixed some edge cases for quantity adjustment. Addresses issue #931.

### CODE REVIEW

File: packages/core/e2e/shop-order.e2e-spec.ts [hunk 1]
```diff
@@ -520,17 +520,56 @@ describe('Shop orders', () => {
...
+
+            // But it should still add the item to the order
+            expect(addItemToOrder!.order.lines[1].quantity).toBe(100);
+
+            const { adjustOrderLine } = await shopClient.query<
+                AdjustItemQuantity.Mutation,
+                AdjustItemQuantity.Variables
+            >(ADJUST_ITEM_QUANTITY, {
+                orderLineId: 'T_8',
+                quantity: 99,
```
Comment (michaelbromley) [inline packages/core/e2e/shop-order.e2e-spec.ts]: I am a bit confused by this part. The test says "adjustOrderLine with quantity > stockOnHand" but here I see that quantity (99) is _less than_ stockOnHand (100). Or am I missing something?

Comment (WilliamMilne) [reply packages/core/e2e/shop-order.e2e-spec.ts]: You're completely right! I have updated that to now be 101 instead and am pushing that code now.

**Comment by michaelbromley** [comment]
I pulled down your branch and played around with it locally. One thing I noticed is that the new test seems to pass event when I revert your changes to `order.service.ts`. Can you confirm whether you see the same?

If so, could you see if you can create a test which fails before the changes, and passes after them?

**Comment by quangpdt** [comment]
The old service is passed with new test is understandable. The incorrect quantity is only returned when we make another query to get the Active order, which is not included in the test case.

@WilliamMilne: Thanks for the fix, I have been looking for the solution for couple days and tried to fix it myself, but clearly it's not covered enough comparing to yours. 

Personally, I found this issue is a critical one and need to be fix asap. Would be great if @michaelbromley could help us out, the PR owner has not given new update for more than a month.

**Comment by WilliamMilne** [comment]
@quangpdt and @michaelbromley - I was able to get a test to fail on `master` and pass on this branch by adding a `GetActiveOrder` query and checking the quantity. On `master`, the quantity was `101`, but on the branch, it was the correct value of `100`. Unfortunately I've hit a small snag and am encountering some failing tests that appear to be failing on both master and the branch: 
image
image

@michaelbromley do you have any suggestions on how to fix these issues so I can push the latest changes? Thanks!!

**Comment by WilliamMilne** [comment]
@michaelbromley - it appears that the customers[0].email issue was solved by deleting the existing e2e data in the folder. 

I was also using an outdated version of node, which was causing the asset server test to crash. I've pushed the latest change now! Sorry about the multiple comments!

**Comment by quangpdt** [comment]
Thanks for the great fix @WilliamMilne!

**Comment by michaelbromley** [comment]
Thank you @quangpdt for your input and @WilliamMilne for your effort to wrap this up. I will review again and all being well, merge this tomorrow.

**Comment by michaelbromley** [comment]
Looks good, thanks all! Will do a patch release this week including this fix.

**Pull request merged**